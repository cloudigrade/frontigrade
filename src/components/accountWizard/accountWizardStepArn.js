import React from 'react';
import { Form, Grid, Icon } from 'patternfly-react';
import { connect, reduxTypes, store } from '../../redux';
import { FormField, fieldValidation } from '../formField/formField';
import Tooltip from '../tooltip/tooltip';

class AccountWizardStepArn extends React.Component {
  state = {
    arn: '',
    arnError: null
  };

  // ToDo: validation for ARN needs to be updated from empty to regex.
  onChangeArn = event => {
    const { value } = event.target;
    const errorMessage = fieldValidation.isEmpty(value) ? 'You must enter a valid ARN' : '';

    this.setState(
      {
        arn: value,
        arnError: errorMessage
      },
      () => this.isStepValid()
    );
  };

  isStepValid() {
    const { arn, arnError } = this.state;
    const stepValid = arnError === '';
    const dispatchType = stepValid
      ? reduxTypes.account.ADD_ACCOUNT_WIZARD_STEP_ARN
      : reduxTypes.account.INVALID_ACCOUNT_WIZARD_STEP_ARN;

    store.dispatch({
      type: dispatchType,
      account: {
        arn
      }
    });
  }

  render() {
    const { arn, arnError } = this.state;

    return (
      <Form horizontal>
        <Form.FormGroup>
          <Grid.Col sm={12}>
            <ul className="no-margin-bottom">
              <li>
                Click <strong>Cloud-Meter-role</strong>.{' '}
                <Tooltip
                  delayShow={100}
                  popover={
                    <ul className="cloudmeter-popover-list">
                      <li>
                        In <strong>IAM</strong> go to the <strong>Roles</strong> tab
                      </li>
                      <li>
                        Click <strong>Cloud-Meter-role</strong>
                      </li>
                    </ul>
                  }
                >
                  <Icon type="pf" name="info" size="large" />
                  <span className="sr-only">Steps when selecting a role</span>
                </Tooltip>
              </li>
              <li>
                Copy the <strong>Role ARN</strong> and paste it in the ARN field.
              </li>
            </ul>
          </Grid.Col>
        </Form.FormGroup>
        <FormField label="ARN" error={arnError} errorMessage={arnError} colLabel={2} colField={10}>
          <Form.FormControl type="text" name="arn" value={arn} placeholder="Enter an ARN" onChange={this.onChangeArn} />
          <Form.HelpBlock>ex: arn:aws:iam::123456789012:role/Cloud-Meter-role</Form.HelpBlock>
        </FormField>
      </Form>
    );
  }
}

AccountWizardStepArn.propTypes = {};

AccountWizardStepArn.defaultProps = {};

const mapStateToProps = state => ({ ...state.accountWizard });

const ConnectedAccountWizardStepArn = connect(mapStateToProps)(AccountWizardStepArn);

export { ConnectedAccountWizardStepArn as default, ConnectedAccountWizardStepArn, AccountWizardStepArn };
