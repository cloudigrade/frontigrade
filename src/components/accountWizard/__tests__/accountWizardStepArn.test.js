import React from 'react';
import configureMockStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import { ConnectedAccountWizardStepArn } from '../accountWizardStepArn';

describe('AccountWizardStepArn Component', () => {
  const generateEmptyStore = (obj = {}) => configureMockStore()(obj);

  it('should render a wizard ARN step', () => {
    const store = generateEmptyStore({ accountWizard: { show: true } });
    const component = shallow(<ConnectedAccountWizardStepArn />, { context: { store } });

    expect(component.dive()).toMatchSnapshot('connected');
  });
});
