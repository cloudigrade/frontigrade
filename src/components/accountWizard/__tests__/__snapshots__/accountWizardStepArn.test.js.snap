// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`connected 1`] = `
<Form
  bsClass="form"
  componentClass="form"
  horizontal={true}
  inline={false}
>
  <FormGroup
    bsClass="form-group"
  >
    <Col
      bsClass="col"
      componentClass="div"
      sm={12}
    >
      <ul
        className="no-margin-bottom"
      >
        <li>
          Click 
          <strong>
            Cloud-Meter-role
          </strong>
          .
           
          <Tooltip
            delayShow={100}
            id={null}
            placement="top"
            popover={
              <ul
                className="cloudmeter-popover-list"
              >
                <li>
                  In 
                  <strong>
                    IAM
                  </strong>
                   go to the 
                  <strong>
                    Roles
                  </strong>
                   tab
                </li>
                <li>
                  Click 
                  <strong>
                    Cloud-Meter-role
                  </strong>
                </li>
              </ul>
            }
            tooltip={null}
            trigger={
              Array [
                "hover",
              ]
            }
          >
            <Icon
              name="info"
              size="large"
              type="pf"
            />
            <span
              className="sr-only"
            >
              Steps when selecting a role
            </span>
          </Tooltip>
        </li>
        <li>
          Copy the 
          <strong>
            Role ARN
          </strong>
           and paste it in the ARN field.
        </li>
      </ul>
    </Col>
  </FormGroup>
  <FormField
    colField={10}
    colFieldClassName={null}
    colLabel={2}
    colLabelClassName={null}
    error={null}
    errorMessage={null}
    id={null}
    label="ARN"
  >
    <FormControl
      bsClass="form-control"
      componentClass="input"
      name="arn"
      onChange={[Function]}
      placeholder="Enter an ARN"
      type="text"
      value=""
    />
    <HelpBlock
      bsClass="help-block"
    >
      ex: arn:aws:iam::123456789012:role/Cloud-Meter-role
    </HelpBlock>
  </FormField>
</Form>
`;
